samples/basic/button/src/main.c

```diff
@@ -13,89 +14,143 @@
 
 #define SLEEP_TIME_MS	1
 
-#if DT_PHA_HAS_CELL(DT_ALIAS(sw0), gpios, flags)
-#define SW0_FLAGS DT_GPIO_FLAGS(DT_ALIAS(sw0), gpios)
-#else
-#define SW0_FLAGS 0
-#endif
+/*
+ * Devicetree helper macro which gets the 'flags' cell from a 'gpios'
+ * property, or returns 0 if the property has no 'flags' cell.
+ */
+
+#define FLAGS_OR_ZERO(node)						\
+	COND_CODE_1(DT_PHA_HAS_CELL(node, gpios, flags),		\
+		    (DT_GPIO_FLAGS(node, gpios)),			\
```

Comment:
Why is it necessary to enclose `DT_GPIO_FLAGS(node, gpios)` in parentheses?  Shouldn't the macro evaluate to an enclosed expression?

(Asking in part because I think I did `(type *)DT_SOMETHING_REG()` a couple weeks ago and got an error because `DT_SOMETHING_REG()` wasn't internally parenthesized.  If we have to wrap the macro calls it's likely to be a cause of problems; if we don't I don't think we should do so in examples.)

Reply:
> Why is it necessary to enclose `DT_GPIO_FLAGS(node, gpios)` in parentheses? Shouldn't the macro evaluate to an enclosed expression?

I don't think that's part of the API guaranteed by DT_GPIO_FLAGS, and COND_CODE_1 of course does require it.

> If we have to wrap the macro calls it's likely to be a cause of problems; if we don't I don't think we should do so in examples

I tend towards keeping it so that users develop the habit of enclosing COND_CODE_1 arguments in parentheses, especially since this sort of sample is likely to be copy/pasted into out of tree applications. But I'm willing to change it if you feel strongly.

Reply:
OK, I see the second argument to `COND_CODE_1` is documented to require parentheses (aka "brackets" for some countries).  I don't have a problem with that, just was unaware.

If I can reproduce the problem where `(type*)DT_FOO_REG_ADDR()` caused grief I'll file an issue on that, but it was probably my error since that idiom seems common in existing code.